### PR TITLE
[szip, openssl] Fix problems with mingw

### LINKED ISSUE #14322: [szip:x64-mingw-dynamic] Reason for `Could not find proper second linker member`
I am playing with my version of the vcpkg, as per this https://github.com/microsoft/vcpkg/issues/10349 issue, to make it bootstrap under MSYS2-MINGW64 without using any Visual Studio related toolchain. I changed the relevant parts of the bootstrap scripts and made the C++ codes compile with g++ etc. Now I have a vcpkg version bootstrapped under MSYS2 running fine. Most of the ports I am interested in are also building just fine, keep in mind that I am using community triplet `x64-mingw-dynamic` and it is not officially supported in the first place. 

But while building `szip` as part of `opencv`:

> ./vcpkg.exe install szip:x64-mingw-dynamic

>Your feedback is important to improve Vcpkg! Please take 3 minutes to complete our survey by running: vcpkg contact --survey
Computing installation plan...
The following packages will be built and installed:
    szip[core]:x64-mingw-dynamic
Detecting compiler hash for triplet x64-mingw-dynamic...
Starting package 1/1: szip:x64-mingw-dynamic
Building package szip[core]:x64-mingw-dynamic...
Could not locate cached archive: C:\Users\burak\AppData\Local\vcpkg\archives\78\789f979763e4fb257ce37380862d2c721a65fb74.zip
-- Using community triplet x64-mingw-dynamic. This triplet configuration is not guaranteed to succeed.
-- [COMMUNITY] Loading triplet configuration from: C:\dev\bin\msys64\opt\vcpkg\triplets\community\x64-mingw-dynamic.cmake
-- Using cached C:/dev/bin/msys64/opt/vcpkg/downloads/szip-2.1.1.tar.gz
-- Cleaning sources at C:/dev/bin/msys64/opt/vcpkg/buildtrees/szip/src/szip-2.1.1-6b767acfa7.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/dev/bin/msys64/opt/vcpkg/downloads/szip-2.1.1.tar.gz
-- Applying patch fix-szip-config-to-set-szip-found.patch
-- Using source at C:/dev/bin/msys64/opt/vcpkg/buildtrees/szip/src/szip-2.1.1-6b767acfa7.clean
-- Configuring x64-mingw-dynamic
-- Building x64-mingw-dynamic-dbg
-- Building x64-mingw-dynamic-rel
-- Performing post-build validation
Could not find proper second linker member

I want to know what could be the reason for `Could not find proper second linker member` message since actual build proceeds without errors. I didn't modify `cofffilereader.cpp` to make it bootstrap under MSYS2 and it was validating other ports just fine. If someone could point me towards the correct direction as to cause of that error msg, I would be glad. 

I found a ticket on a similar issue: 
https://github.com/microsoft/vcpkg/pull/10398

but I don't know how to make use of it.

I also used `dumpbin` myself on the generated lib file. It indeed shows:
>36C size
correct header end
    34 public symbols

for `dumpbin /linkermember:1 szip_D.lib` but not for `dumpbin /linkermember:2 szip_D.lib`. The other lib files I generated with my `vcpkg` don't have such problems. Is it maybe `szip` is not viable to building as a dynamic library?

### PR DESCRIPTION
**Pull request by longnguyen2004**
**Describe the pull request**

- What does your PR fix? Fixes #14322 
[szip] Disable architecture checks
[openssl] Use the unix portfile with mingw triplets
- Which triplets are supported/not supported? Have you updated the CI baseline?
All mingw triplets
- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

File: ports/szip/portfile.cmake [hunk 1]
```diff
@@ -1,3 +1,7 @@
+if(VCPKG_TARGET_IS_MINGW)
+    set(VCPKG_POLICY_SKIP_ARCHITECTURE_CHECK enabled)
+endif()
```
Comment (Neumann-A) [inline ports/szip/portfile.cmake:3]: no this setting is a triplet setting and belongs in the triplet.

Comment (longnguyen2004) [reply ports/szip/portfile.cmake:3]: I'll move it to the toolchain file then, is it OK?

Comment (Neumann-A) [reply ports/szip/portfile.cmake:3]: it belong into the triplet not the toolchain.

> The problem is, most ports are OK without this policy, except for szip for some reason.

shared vs static library maybe?

Comment (longnguyen2004) [reply ports/szip/portfile.cmake:3]: Every port I've tested (libpng, sfml, openssl, boost, and many others) works fine without this, except for szip. I'll take a look at the dll file to see if there's anything wrong.

Comment (longnguyen2004) [reply ports/szip/portfile.cmake:3]: Nothing strange in the dll file either, guess I'll just move it into the triplets then.

**Review by PhoebeHui — COMMENTED** [review]
LGTM, @strega-nil, could you help further review?

**Pull request merged**